src/redux/books/booksSlice.js

```diff
@@ -51,10 +51,10 @@ export const BookSlice = createSlice({
         state.error = action.error.message;
       })
       .addCase(addBook.fulfilled, (state, action) => {
-        state.books.push(action.payload);
+        Object.assign(state.books, action.payload);
       })
       .addCase(removeBook.fulfilled, (state, action) => {
-        state.books = state.books.filter((book) => book.item_id !== action.payload);
+        delete state.books[action.payload];
       });
```

Comment:
- [ ] Good job making the changes @KUBAHO3, But following your change I have an issue adding books. This also prevents me from checking if your remove book functionality is working fine. Please verify your remove and add book functionalities are working fine before the next review request. 
[Screencast from 10-05-2023  8:20:57 ከሰዓት.webm](https://github.com/KUBAHO3/book-store/assets/69077061/3857b386-b9b4-44f0-a9e9-5a162f678f34)

Reply:
Even though the Add new book is working fine in my system let me try a new approach!